blocks/form/components/repeat/repeat.js

```
@@ -38,46 +85,36 @@ function createButton(label, icon) {
 }
 
-export function insertRemoveButton(fieldset, wrapper, form) {
-  const label = fieldset.dataset?.repeatDeleteButtonLabel || 'Delete';
-  const removeButton = createButton(label, 'remove');
-  removeButton.addEventListener('click', () => {
-    fieldset.remove();
-    wrapper.querySelector('.item-add').setAttribute('data-visible', 'true');
-    wrapper.querySelectorAll('[data-repeatable="true"]').forEach((el, index) => {
-      update(el, index, wrapper['#repeat-template-label']);
-    });
-    const event = new CustomEvent('item:remove', {
-      detail: { item: { name: fieldset.name, id: fieldset.id } },
-      bubbles: false,
-    });
-    form.dispatchEvent(event);
-  });
-  fieldset.append(removeButton);
-}
+/**
+ * Updates the visibility of add/remove buttons based on min/max constraints.
+ * @param {HTMLElement} wrapper - The repeat wrapper element
+ */
+function updateButtonVisibility(wrapper) {
+  const instances = wrapper.querySelectorAll('[data-repeatable="true"]');
+  const count = instances.length;
+  const min = parseInt(wrapper.dataset.min || 0, 10);
+  const max = parseInt(wrapper.dataset.max || -1, 10);
 
-export const add = (wrapper, form, actions) => (e) => {
-  const fieldset = wrapper['#repeat-template'];
-  const max = wrapper.getAttribute('data-max');
-  const min = wrapper.getAttribute('data-min');
-  const childCount = wrapper.children.length - 1;
-  const newFieldset = fieldset.cloneNode(true);
-  newFieldset.setAttribute('data-index', childCount);
-  update(newFieldset, childCount, wrapper['#repeat-template-label']);
-  if (childCount >= +min) {
-    insertRemoveButton(newFieldset, wrapper, form);
-  }
-  if (+max !== -1 && +max <= childCount + 1) {
-    e.currentTarget.setAttribute('data-visible', 'false');
+  const addButton = wrapper.querySelector('.item-add');
+  if (addButton) {
+    if (max !== -1 && count >= max) {
+      addButton.style.display = 'none';
+    } else {
+      addButton.style.display = '';
+    }
```

Let's add a data attribute in the wrapper when it reaches the max/min and based on that style the component. Default styling would be hide them, but customisations should allow keeping the buttons disabled.

---

Done 👍